### PR TITLE
Escape spaces in project path

### PR DESCRIPTION
**Pull request by cstroe**
This allows scalatest-maven-plugin to run projects that have a space in their path.

In the process, I had to update the integration tests so that they run correctly (the lift integration test wasn't compiling its scala files) and properly verify that scalatest actually ran tests.

### CODE REVIEW

**Comment by cstroe** [comment]
Fixes #38

File: src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java [hunk 1]
```diff
@@ -426,12 +427,37 @@ private ClassLoader classLoader() {
     }
 
     private List<String> runpath() {
```
Comment (cstroe) [inline src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java]: The main fix is in this method.

**Comment by cstroe** [comment]
@bvenners @cheeseng Reaching out to you as you seem to be the top contributors to this repository.  I would appreciate if someone would review and merge this PR.  Let me know if there are any questions.

**Comment by cstroe** [comment]
Adding an integration test for the fix in this PR.  (They were removed in an earlier PR: https://github.com/scalatest/scalatest-maven-plugin/pull/40)

**Comment by cstroe** [comment]
@gcberger I tested version 2.0.0-SNAPSHOT and it still doesn't build project that have spaces in their path.  This PR fixes the issue, and I would appreciate it if it were considered for release in 2.0.0.

**Comment by bvenners** [comment]
@cstroe you're PR came in after we had gone through the PRs for 2.0.0, which is why it didn't get in there. Doesn't mean we can't consider it. I'll take a look at your PR tomorrow while people are trying out the SNAPSHOT.

**Comment by cstroe** [comment]
@bvenners Any chance in merging this?

**Comment by brosander** [comment]
We just ran into this issue, chance of getting fix merged?

@bvenners status?

**Comment by bvenners** [comment]
@cstroe Sorry for the long delay on this PR. Do you mind signing our CLA? It is here:

https://www.artima.com/cla/contributorsLicenseAgreement.pdf

**Comment by cstroe** [comment]
@bvenners I have signed the CLA and emailed it.

**Comment by mlekena** [comment]
Just ran into this issue. Is this still on track to be merged? What would a patch be to get around this issue in the mean time??

File: src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java [hunk 2]
```diff
@@ -426,12 +427,37 @@ private ClassLoader classLoader() {
...
+        checkRunpathArgument(outputDirectory);
```
[the remark this answers is not in the document]
Comment (katrinsharp) [reply src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java:430]: Also, please resolve conflicts. Thank you.

[the remark this answers is not in the document]
Comment (cstroe) [reply src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java:434]: Fixed warnings and merge conflict.

[the remark this answers is not in the document]
Comment (cstroe) [reply src/main/java/org/scalatest/tools/maven/AbstractScalaTestMojo.java:434]: The "runpath argument" is the value passed to the `-R` flag of the ScalaTest runner.

The warning "Runpath directory does not exist" gets printed when a non-existent directory is passed to the ScalaTest runner.  This should not happen during normal operation, and may indicate a bug in our code.

**Comment by katrinsharp** [comment]
LGTM

**Pull request merged**